### ISSUE TITLE
Doxygen 1.9.7 - Java interface's methods repeated in the implementation file documentation

### ISSUE THREAD

**Issue by MarkShackman** [doxygen/doxygen#10157]
[doxygen 1.9.7]

With two separate files containing a Java interface and the corresponding implementation, the generated API summary for the implementation class contains the methods of both the interface and implementation.

How can I remove the duplication (ideally without filtering the code)?

Thanks.

[JavaInterfaceImplementationMethodDuplication.zip](https://github.com/doxygen/doxygen/files/11908383/JavaInterfaceImplementationMethodDuplication.zip)

![image](https://github.com/doxygen/doxygen/assets/23169476/407da1a2-6887-4449-87fc-a4dd11e19dd1)

**Comment by albert-github** [comment]
I checked the 1.9.7 version and saw in the file classcom_1_1company_1_1_implementation.html the double member functions
Checking against the current master (1.9.8 (396688347198e076597050259f4835853bdc3b44) showed:

![image](https://github.com/doxygen/doxygen/assets/5223533/4eef4efe-a725-4f83-be3c-80e3d368df24)

So I think the problem is fixed but not released.

**Comment by MarkShackman** [comment]
Thanks; yes, confirmed as fixed in 1.9.8 development.  Is there an (easy) workaround available until that's released?

**Comment by albert-github** [comment]
@MarkShackman 
I'm not aware of a workaround,
The fix looks like to come from the fix for issue #10081
```
Commit: c58aa9e5d9bd921b34236f171a82c47d7f160517 [c58aa9e]
Date: Sunday, May 28, 2023 1:33:22 PM

issue #10081 [C++] Inherited virtual functions are not listed as being inherited in child class
```

more precise the line:
from
```
      writePlainDeclarations(ol,inGroup,cd,nd,fd,gd,0,0,inheritId);
```
to
```
      writePlainDeclarations(ol,inGroup,cd,nd,fd,gd,0,inheritedFrom,inheritId);
```
in memberlist.cpp  (other are just print corrections or a change that is later reverted)

**Comment by MarkShackman** [comment]
My workaround is to wrap the contents of the interface in a doxygen group (with a blank @name). Has the trade-off of not sorting the method summary alphabetically even if SORT_BRIEF_DOCS =YES, but that's acceptable until 1.9.8 is released.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.8.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
